tests/core/_orchestrator/_dispatcher/test_development_job_dispatcher.py

```
@@ -0,0 +1,63 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import traceback
+from unittest.mock import patch
+
+from taipy.core import JobId
+from taipy.core._orchestrator._orchestrator_factory import _OrchestratorFactory
+from taipy.core.job.job import Job
+from taipy.core.task._task_manager_factory import _TaskManagerFactory
+from taipy.core.task.task import Task
+
+
+def nothing(*args):
+    return
+
+
+def create_task():
+    task = Task("config_id", {}, nothing, [], [])
+    _TaskManagerFactory._build_manager()._set(task)
+    return task
+
+
+def test_dispatch_executes_the_function_no_exception():
+    task = create_task()
+    job = Job(JobId("job"), task, "s_id", task.id)
+    dispatcher = _OrchestratorFactory._build_dispatcher()
+
+    with patch("taipy.core._orchestrator._dispatcher._task_function_wrapper._TaskFunctionWrapper.execute") as mck:
+        mck.return_value = []
+        dispatcher._dispatch(job)
+
+        mck.assert_called_once()
+
+    assert job.is_completed()
+    assert job.stacktrace == []
+
+
+def test_dispatch_executes_the_function_with_exceptions():
+    task = create_task()
+    job = Job(JobId("job"), task, "s_id", task.id)
+    dispatcher = _OrchestratorFactory._build_dispatcher()
+    e_1 = Exception("test")
+    e_2 = Exception("test")
+
+    with patch("taipy.core._orchestrator._dispatcher._task_function_wrapper._TaskFunctionWrapper.execute") as mck:
+        mck.return_value = [e_1, e_2]
```

Instead of mocking the return value, how about executing a function that raise the exception?

---

This is possible, but you must create a function and the corresponding task.
In the end, we don't need to execute anything. We are just asserting that execute is called, and when exceptions are returned, the job is correctly updated. 
It does not bring any value to me to run a function.